blackbox/docs/sql/administration/privileges.txt

```diff
@@ -0,0 +1,147 @@
+.. highlight:: psql
+.. _administration-privileges:
+
+=======================
+Cluster Wide Privileges
+=======================
+
+.. note::
+
+    Privileges are only part of the CrateDB `Enterprise Edition`_.
+    When the CrateDB Enterprise Edition is disabled, there will be
+    no user privilege checks, and every statement will be executed
+    without the validation of privileges.
+
+
+``Privilege Types``
+====================
+
+When a user is created, he doesn't have any privilege. The superuser can then
+grant privileges to this user.
+
+.. note::
+
+    Superusers are allowed to execute any statement without any privilege
+    checks. Currently only ``DQL``, ``DML`` and ``DDL`` privileges can be
+    granted. Any statements which are not allowed with those privileges,
+    such as ``GRANT`` and ``REVOKE``, can only be issued by a superuser.
+
+    At the moment, only the initial superuser ``crate`` is available, it is
+    not possible to create any other superusers, or manage their privileges.
+
+
+User privileges are applied on the whole cluster. There is no option to ``GRANT``
+or ``REVOKE`` privileges on object level, such as table or schema.
+
+``DQL``
+.......
+
+Granting ``Data Query Language (DQL)`` privilege to a user, indicates that this
+user is allowed to execute ``SELECT``, ``SHOW`` and ``SET SESSION`` statements,
+as well as using the available user defined functions.
+
+
+``DML``
+.......
+
+Granting ``Data Manipulation Language (DML)`` privilege to a user, indicates
+that this user is allowed to execute ``INSERT``, ``UPDATE`` and ``DELETE``
+statements. In practice any non trivial ``DML`` statement  will require
+``DQL`` privilege, for example ``INSERT FROM`` with a subquery.
+
+
+``DDL``
+.......
+
+Granting ``Data Definition Language (DDL)`` privilege to a user, indicates that
+this user is allowed to execute ``CREATE``, ``DROP`` and ``ALTER`` statements.
+
+
+``GRANT``
+=========
+
+.. hide:
+
+    cr> CREATE user wolfgang;
+    CREATE OK, 1 row affected (... sec)
+
+    cr> CREATE user will;
+    CREATE OK, 1 row affected (... sec)
+
+To grant a privilege to an existing user we use the :doc:`/sql/reference/grant`
+SQL statement, for example::
+
+    cr> GRANT DML TO wolfgang;
+    GRANT OK, 1 row affected (... sec)
+
+
+If a user with the username specified in the SQL statement does not exist the
+statement returns an error::
+
+    cr> GRANT DQL TO layla;
+    UserDoesNotExistException ...
+
+
+To grant ``ALL PRIVILEGES`` to user will, we can use the following syntax::
+
+    cr> GRANT ALL PRIVILEGES TO will;
+    GRANT OK, 2 rows affected (... sec)
+
+``ALL PRIVILEGES`` implies all the currently available privileges,
```

Comment:
using "ALL..."  is a shortcut to grant all the currently grantable privileges to a user, namely  ...